apiserver/facades/agent/provisioner/service.go

```diff
@@ -72,6 +72,11 @@ type MachineService interface {
 	// GetMachineUUID returns the UUID of a machine identified by its name.
 	// It returns a MachineNotFound if the machine does not exist.
 	GetMachineUUID(ctx context.Context, name coremachine.Name) (string, error)
+	// SetLXDProfiles adds the list of LXD profile names to the lxd_profile table
+	// for the given machine.
+	// [machineerrors.MachineNotFound] will be returned if the machine does not
+	// exist.
+	SetLXDProfiles(ctx context.Context, mUUID string, profileNames []string) error
```

Comment:
This would be a lot better if machine had a UUID type. Then we wouldn't need to use this Hungarian type annotation.